app/src/main/java/com/termux/api/apis/BluetoothScanAPI.java

```diff
@@ -0,0 +1,122 @@
+package com.termux.api.apis;
+
+import android.bluetooth.BluetoothAdapter;
+import android.bluetooth.BluetoothDevice;
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.content.IntentFilter;
+import android.util.JsonWriter;
+
+import com.termux.api.util.ResultReturner;
+import com.termux.shared.logger.Logger;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Objects;
+
+public class BluetoothScanAPI {
+
+    private static final ArrayList<BluetoothDevice> devices = new ArrayList<>();
+    private static final String LOG_TAG = "BluetoothScanAPI";
+    private static BluetoothAdapter adapter;
+    private static boolean scanning = false;
+
+    private static final BroadcastReceiver receiver = new BroadcastReceiver() {
+        @Override
+        public void onReceive(Context context, Intent intent) {
+            if (Objects.equals(intent.getAction(), BluetoothDevice.ACTION_FOUND)) {
+                BluetoothDevice device = intent.getParcelableExtra(BluetoothDevice.EXTRA_DEVICE);
+                if (device != null) {
+                    devices.add(device);
+                }
+            }
+        }
+    };
+
+    public static void onReceive(final Context context, final Intent intent) {
+        Logger.logDebug(LOG_TAG, "onReceive");
+
+        ResultReturner.returnData(context, intent, new ResultReturner.ResultJsonWriter() {
+            @Override
+            public void writeJson(JsonWriter out) throws Exception {
+                String mode = intent.getStringExtra("mode");
+                if (mode == null) return;
+                out.beginObject();
+                switch (mode.toLowerCase()) {
+                    case "start":
+                        if (start(context)) {
+                            out.name("error").value(false);
+                        } else {
+                            out.name("error").value(true);
+                            out.name("reason").value("Already running!");
```

Comment:
Instead of having two keys, how about `error` only being present if it has an error message and returning an empty object on success? There's [approximate precedent for that](https://github.com/termux/termux-api/blob/cff225e41e963267a8e5fdd59a83b5fe224dcb5d/app/src/main/java/com/termux/api/apis/NfcAPI.java#L52-L53) in the NFC API.